### PR TITLE
Replace Travis with GitHub Actions

### PR DESCRIPTION
**Pull request by nllong**
#### Any background context you want to provide?
Travis is now charging for projects once you run out of credits.

#### What does this PR accomplish?
* Move over to github actions

#### How should this be manually tested?
* Check the github actions on this PR

#### What are the relevant tickets?
#243

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+          tox -e python
+      -
+        name: Job Failed
+        if: ${{ failure() }}
+        run: |
+          echo "Maybe these logs will help?"
+          free -tm
+          ls -alt $GITHUB_WORKSPACE
+          echo "============================================ stdout.log ========================================="
+          find $GITHUB_WORKSPACE -type f -name 'stdout.log' -print | while read filename; do
```
Comment (macintoshpie) [inline .github/workflows/ci.yml:52]: nice 👍

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+          # Install modelicafmt (hard code the version for now)
+          curl -SLO https://github.com/urbanopt/modelica-fmt/releases/download/v0.1-pr.3/modelica-fmt_0.1-pr.3_Linux_x86_64.tar.gz
+          tar xzf modelica-fmt_0.1-pr.3_Linux_x86_64.tar.gz
+          chmod +x modelicafmt
+          sudo mv modelicafmt /usr/local/bin/
+          pip install --upgrade pip
+          pip install tox==3.13.2
+          pip install requests
+      -
+        name: Run pre-commit
```
Comment (macintoshpie) [inline .github/workflows/ci.yml:80]: I guess this will be run twice (b/c of matrix) but easier than splitting it out.

**Review by macintoshpie — APPROVED** [review]
LGTM 👍 ⚡ 💥

File: .github/workflows/ci.yml (hunk 2 above)
Comment (nllong) [reply .github/workflows/ci.yml:80]: oh, oops, right.

**Pull request merged**